### PR TITLE
Adds support for Dropwizard 5 metrics.

### PR DESCRIPTION
**Pull request by pennello**
Hi @brettwooldridge ,

Here is a PR that adds support for Dropwizard 5 metrics alongside existing support for Codahale and Micrometer.

I tried to keep the code style consistent with the existing code, and with new tests, sought to keep common as many tests as possible for Codahale and Dropwizard 5, which are of course very similar, but also frustratingly different when shared in the same codebase.  (I did this by having a test abstract base class and parameterizing the metric registry type.)

The existence of `IMetricsTracker` made adding Dropwizard 5 support relatively straightforward.  But I didn't see a similar abstraction for health checking.  Hence I didn't add support for that yet.  I figured that this is still a strict improvement and would thus still be worth submitting to you.

I also made some library upgrades in order to get tests passing in a Java 18 environment.

Please let me know what you think, or if you think this might need any revisions before it's appropriate to get into `dev`.

---

For context, my company has been working on a Jetty 11 upgrade, which has necessitated a number of other nontrivial and sometimes backwards-incompatible upgrades, including Resteasy, Jakarta (over `javax`), and Dropwizard.  Hence, to maintain the useful Hikari metrics, we wanted to add support here for Dropwizard 5.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -9,14 +9,15 @@
...
+      <docker.maven.plugin.fabric8.version>0.40.2</docker.maven.plugin.fabric8.version>
       <felix.bundle.plugin.version>5.1.1</felix.bundle.plugin.version>
-      <felix.version>6.0.1</felix.version>
+      <felix.version>7.0.5</felix.version>
       <hibernate.version>5.4.24.Final</hibernate.version>
       <javassist.version>3.27.0-GA</javassist.version>
       <jndi.version>0.11.4.1</jndi.version>
       <maven.release.version>2.5.3</maven.release.version>
       <metrics.version>3.2.5</metrics.version>
+      <metrics5.version>5.0.0-rc11</metrics5.version>
```
Comment (pennello) [inline pom.xml]: You might naturally question managing this dependency version on an RC, but [they've been RC'ing this for over two years now][1], so I _think_ this is appropriate.

[1]: https://search.maven.org/artifact/io.dropwizard.metrics5/metrics-core

File: pom.xml [hunk 2]
```diff
@@ -452,6 +460,7 @@
                      javax.sql.rowset.spi,
                      com.codahale.metrics;resolution:=optional,
                      com.codahale.metrics.health;resolution:=optional,
+                     io.dropwizard.metrics5;resolution:=optional,
```
Comment (pennello) [inline pom.xml:473]: I wasn't quite sure what this was, but given the pre-existing `codahale` entries, I figured this was appropriate to add to help with your release process.

File: src/test/java/com/zaxxer/hikari/pool/CodahaleMetricsTest.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeFalse;
+
+/**
+ * Test HikariCP/CodaHale metrics integration.
+ *
+ * @author Brett Wooldridge
+ */
+public class CodahaleMetricsTest extends TestMetricsBase<MetricRegistry>
```
Comment (pennello) [inline src/test/java/com/zaxxer/hikari/pool/CodahaleMetricsTest.java:51]: This has your pre-existing tests from the formerly-named `TestMetrics` that couldn't be made generic.

File: src/test/java/com/zaxxer/hikari/pool/Dropwizard5MetricsTest.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+import static com.zaxxer.hikari.pool.TestElf.newHikariDataSource;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeFalse;
+
+/**
+ * Test HikariCP/Dropwizard 5 metrics integration.
+ */
+public class Dropwizard5MetricsTest extends TestMetricsBase<MetricRegistry>
```
Comment (pennello) [inline src/test/java/com/zaxxer/hikari/pool/Dropwizard5MetricsTest.java:30]: Correspondingly, this has similarly-patterned tests, but for Dropwizard 5.

File: src/test/java/com/zaxxer/hikari/pool/TestMetrics.java [hunk 5]
```diff
@@ -16,289 +16,18 @@
...
-import com.zaxxer.hikari.util.UtilityElf;
+import static com.zaxxer.hikari.pool.TestElf.newHikariDataSource;
 
 /**
- * Test HikariCP/CodaHale metrics integration.
+ * Test HikariCP metrics integration.
  *
  * @author Brett Wooldridge
  */
 public class TestMetrics
```
Comment (pennello) [inline src/test/java/com/zaxxer/hikari/pool/TestMetrics.java:29]: Logically, this got split into to `TestMetricsBase` and `CodahaleMetricsTest`.  Retained here is the single Codahale/Dropwizard-independent test (for the fake registry).

**Comment by pennello** [comment]
> Some checks were not successful
> 1 failing and 1 successful checks
> Travis CI - Pull Request Failing after 1m — Build Failed

> `The command "mvn package -Dmaven.javadoc.skip=true -V -B" exited with 1.`

Rats.  When I run that locally on my branch, it works.  I'm not sure what's wrong.  I didn't see further logging from Travis.  Am I missing something?

**Comment by pennello** [comment]
Hi @brettwooldridge , just a friendly ping on this.

**Comment by lfbayer** [comment]
@pennello, The travis-ci build for this is failing, any idea why?

**Comment by pennello** [comment]
@lfbayer Thanks for the reply.

I checked the Travis logs and saw only the following:

> `The command "mvn package -Dmaven.javadoc.skip=true -V -B" exited with 1.`

Is there a way to get more diagnostic information out of the CI build?  When I run that locally on my branch, it works.  I've got to be missing something simple.

File: pom.xml (hunk 1 above)
Comment (pennello) [reply pom.xml]: This is _still_ the case as of June 2023.  I advanced the RC version to 17 as of a51f146a87ca5ac9702a87b0c79b716b3429c226.

**Comment by pennello** [comment]
In the hopes of the Travis run succeeding on a re-try, I re-pushed as of a51f146a87ca5ac9702a87b0c79b716b3429c226.  It passed, but I don't see Travis anymore.  Only Snyk.

However, there were conflicts.  I rebased, and also added a module declaration for Dropwizard 5 as of 3f377c0bda615f9abb2efacd2d67766997f3d2f8.

**Pull request merged**